model/task_lifecycle.go

```diff
@@ -132,6 +132,10 @@ func SetActiveState(t *task.Task, caller string, active bool) error {
 		if err := UpdateDisplayTaskForTask(t); err != nil {
 			return errors.Wrap(err, "problem updating display task")
 		}
+	} else {
```

Comment:
Sorry for not catching this: I think we should do this regardless of IsPartOfDisplay; if the execution task status changed, the display task status will also have changed, so the build/version status will have changed.